### ISSUE TITLE
Create tests

### ISSUE THREAD

**Issue by Tempel** [snegovick/bcam#136]
Automated tests are very helpful in maintaining a code base.  All of BCAM's geometry calculations are well-suited for testing; I/O and GUI code is a little more difficult to test, but still very possible.

@snegovick, I'll start working on this, but I want your input before I get started.  I'd like to write the tests using [pytest](http://pytest.org/), but that would add another external dependency.  If you'd prefer to keep everything simpler, I could instead use the standard library's [unittest package](https://docs.python.org/2/library/unittest.html).  Let me know what you think.

**Comment by snegovick** [comment]
Hm I dont mind adding pytest if this dependency is only needed during dev time and can be omitted during package build time. I mean its ok for developers, but it shouldnt be required for end users. If this can be done, then im ok with pytest.

**Comment by Tempel** [comment]
Yes, this would only be needed to run tests, not to run the program.  I'll go ahead with pytest then.  Thanks.

**Comment by Tempel** [comment]
I've put the framework in place for generating a development environment and automatically running tests.  Details on use are in [CONTRIBUTING.md](https://github.com/snegovick/bcam/blob/v0.3/CONTRIBUTING.md).  Now we just need to write those tests :)

**Comment by snegovick** [comment]
Thank you, thats quite impressive !

**Comment by snegovick** [comment]
Will later add test-writting tasks per module
